communication/mavlinkparameterserver.cpp

```diff
@@ -15,16 +20,26 @@ MavlinkParameterServer::MavlinkParameterServer(std::shared_ptr<mavsdk::ServerCom
     mMavsdkParamServer->provide_param_int("MIS_TAKEOFF_ALT", 0);
 }
 
+void MavlinkParameterServer::createSingleton(std::shared_ptr<mavsdk::ServerComponent> serverComponent)
+{
+    if(mInstancePtr)
+        qDebug() << "Parameterserver singleton already created";
+    else
+        mInstancePtr = new MavlinkParameterServer(serverComponent);
```

Comment:
I don't think that it is critical here, but for every 'new' I would like to see a 'delete' otherwise the destructor will not be called anywhere. https://doc.qt.io/qt-5/qscopedpointer.html or std::unique_ptr might be a good idea here (hand out raw pointer to others but control object lifetime internally).